relation_engine/database/test/batch_time_travel_test.py

```diff
@@ -82,4 +82,167 @@ def test_save_vertex(arango_db):
+                   'science': 'yes!'}
+
+    ret = att.get_vertex('myid1', 600)
+    assert ret == {'_key': 'myid1_load-ver1',
+                   'created': 500,
+                   'expired': 9007199254740991,
+                   'first_version': 'load-ver1',
+                   'id': 'myid1',
+                   'last_version': 'load-ver1',
+                   'science': 'yes!'}
 
+def _setup_set_node_expired(arango_db):
+
+    vert_col_name = 'verts'
+    edge1_col_name = 'edges1'
+    edge2_col_name = 'edges2'
+    vert_col = arango_db.create_collection(vert_col_name)
+    fake_col = arango_db.create_collection('fake')
+    edge1_col = arango_db.create_collection(edge1_col_name, edge=True)
+    edge2_col = arango_db.create_collection(edge2_col_name, edge=True)
+
+    vert_col.import_bulk([{'_key': '1', 'id': 'foo', 'created': 100, 'expired': 9000},
+                          {'_key': '2', 'id': 'bar', 'created': 100, 'expired': 9000},
+                          ])
+    fake_nodes = [{'_key': '1', 'id': 'bat', 'created': 100, 'expired': 9000},
+                  {'_key': '2', 'id': 'baz', 'created': 100, 'expired': 9000},
+                  ]
+    fake_col.import_bulk(fake_nodes)
+    
+    edges1 = [{'_key': 'e1_1', '_from': 'verts/1', '_to': 'fake/1',
+               'created': 100, 'expired': 9000},
+              {'_key': 'e1_2', '_from': 'fake/2', '_to': 'verts/1',
+               'created': 100, 'expired': 9000},
+              {'_key': 'e1_3', '_from': 'verts/2', '_to': 'fake/1',
+               'created': 100, 'expired': 9000},
+              {'_key': 'e1_4', '_from': 'fake/2', '_to': 'verts/2',
+               'created': 100, 'expired': 9000}, 
+              ]
+
+    edge1_col.import_bulk(edges1)
+
+    edges2 = [{'_key': 'e2_1', '_from': 'verts/1', '_to': 'fake/1',
+               'created': 100, 'expired': 9000},
+              {'_key': 'e2_2', '_from': 'fake/2', '_to': 'verts/1',
+               'created': 100, 'expired': 9000},
+              {'_key': 'e2_3', '_from': 'verts/2', '_to': 'fake/1',
+               'created': 100, 'expired': 9000},
+              {'_key': 'e2_4', '_from': 'fake/2', '_to': 'verts/2',
+               'created': 100, 'expired': 9000}, 
+              ]
+
+    edge2_col.import_bulk(edges2)
+
+    return edges1, edges2
+
+def test_set_node_expired_single_vertex(arango_db):
+    edges1, edges2 = _setup_set_node_expired(arango_db)
+    att = ArangoBatchTimeTravellingDB(arango_db, default_vertex_collection='verts')
+
+    att.expire_vertex('1', 500)
+
+    ret = att.get_vertex('foo', 200)
+    assert ret == {'_key': '1', 'id': 'foo', 'created': 100, 'expired': 500}
+    ret = att.get_vertex('bar', 200)
+    assert ret == {'_key': '2', 'id': 'bar', 'created': 100, 'expired': 9000}
+
+    _check_no_fake_changes(arango_db)
+
+    _check_docs(arango_db, edges1, 'edges1')
+    _check_docs(arango_db, edges2, 'edges2')
+
+def test_set_node_expired_vert_and_edges_one_collection(arango_db):
+
```

Comment:
Can you add comments in each of these test with a plain english description of what this is testing?

Reply:
Done